docs/client-provisioning-methods.adoc

```diff
@@ -0,0 +1,40 @@
+Every time the device attempts to connect, the server verifies that the device credentials are signed by the same CA that you originally installed on the server.
+
+== Choosing a provisioning method
+
+The type of provisioning that you choose depends on your requirements. If you're just testing and want to get started quickly, provisioning with shared credentials is fine. It is easier to set up but is less secure. Because server self-signs the credentials that it issues to devices, the devices have no way of verifying the integrity of the server. This level of security is not suitable for a production environment. The device should always be able to verify that is communicating with a genuine OTA Connect server. Once you move to production, you should look at some form of provisioning that uses device certificates. For more security, use a Hardware Security Module (HSM) to store the device certificates.
+
+=== Configure a Device to Provision with Shared Credentials
+
+To configure a device to provision with shared credentials, follow these steps:
+
+. Download a https://docs.ota.here.com/quickstarts/generating-provisioning-credentials.html[provisioning key].
+. Edit the following `.toml` configuration file:
++
+link:https://github.com/advancedtelematic/aktualizr/blob/master/config/sota_autoprov.toml[config/sota_autoprov.toml]
+. Update the configuration file with the path to the zip file that you saved in the first step and start Aktualizr on the device.
++
+** When aktualizr starts, it uses the provisioning key to register with the device gateway HTTPS API. 
+** The server registers the new device and returns a PKCS#12 archive containing the real credentials for the device. This archive contains the root CA certificate and client certificate.
+** The device unpacks the archive into the paths that are specified in the `[storage]` section of the config.
+//MC: ?? my [storage] just says "sqlite"...is that a path?
+** The device uses this private key and certificate for all further communication with the server.
+
+=== Configure a Device to Provision with Device Credentials
+
+To configure a device to provision with device credentials, you need a root CA certificate obtained from a trusted certificate authority.
+
+* If you don't have a root CA certificate yet, you can https://docs.ota.here.com/prod/generate-and-install-a-root-certificate.html[generate and install a self-signed root CA certificate] or link:./enable-implicit-provisioning.html#simulate-implicit-provisioning-for-testing[simulate provisioning with device credentials].
+* When you're ready to move to production, follow the procedure to link:./enable-implicit-provisioning.html#enable-implicit-provisioning-with-a-hardware-security-module-hsm[use a Hardware Security Module to provision with device credentials]
```

Comment:
Link is broken, and using an HSM is optional. I'd recommend it, but I can't guarantee that's what all users want or need.

Reply:
Still broken, and my point still stands.